### DISCUSSION TITLE
对于封号风险的应对思路

### DISCUSSION THREAD

**Discussion by borntree** [xisuo67/XHS-Spider#85] [Q&A] [answered]
有个小程序叫小红图去水印，可以不登录就保存图片，保存图片都是webp和jpg格式的。
像素我对比了一下和本项目是一样的，唯一的缺点是无法一键保存。
我在想这个东西是用的僵尸账号收集素材吗？
还是说有什么方法可以规避登陆或封号风险？
亦或是特意不加一键保存所有作品的功能来规避封号风险？（这里可以想出小红书的检测方法可能是用户是否在很短的时间内访问了大量作品，是否可以给本项目加一个定时器来下载作品，从而规避掉封号风险呢？）

**Comment by xisuo67** [comment]
目前不登录也是可以下载图片的，前提是需要知道单条笔记的url

**Comment by xisuo67** [comment]
将问题转入讨论贴中，大家可以集思广益

**Comment by Unka-Malloc** [comment, marked as answer]
小红书确实有检测用户是否在很短时间内访问了大量作品的方法. 

由于发送请求始终需要包含Cookie, 而这个Cookie和用户账号高度绑定, 所以应该是需要大量僵尸账号没错. 值得注意的是, 即使绕过了客户端本地的反爬检测机制, 在服务器上仍然有一个基于流量的检测机制. 这个流量限制触发很容易, 但是并非每时每刻都在检查, 而是周期性的检查并且封号. 也就是说, 只要记得切换Cookie并且保持每个账号的QPS较低, 就可以不触发封号.

**Reply by xisuo67** [reply]
哇！这个回复靠谱，看来得加个cookie池，用于每次请求随机取cookie，来绕过封号的问题了

**Reply by xisuo67** [reply]
增加另外一个功能，多账号扫码登录或账号密码登录，用来记录每个账号登录后的cookie信息，加入到cookie池，请求接口的时候随机取这些账号的cookie信息，这样应该就能解决封号的问题

**Comment by CN-make** [comment]
保持一定程度的賬號活躍就好啦，基本每天都在採集也很少出現封號情況，要是能加上關鍵詞檢索下載，我想贊助當回金主 哈哈哈

**Reply by xisuo67** [reply]
@CN-make 短时间内不太可能加上这个功能，没有太多时间维护这个项目
